doc/config.rst

```diff
@@ -119,6 +119,37 @@ The following example shows how to use all ``PhpCsFixer`` rules but without the
         ->setFinder($finder)
     ;
 
+If you need to disable or reconfigure a rule for specific files, you can use the ``setFilterFixerByFile`` method:
+
+.. code-block:: php
+
+    <?php
+
+    $finder = (new PhpCsFixer\Finder())
+        ->in(__DIR__)
+    ;
+
+    return (new PhpCsFixer\Config())
+        ->setRules([
+            'array_syntax' => ['syntax' => 'short'],
+            'strict_param' => true,
+        ])
+        ->setFinder($finder)
+        ->setFilterFixerByFile(static function (PhpCsFixer\Fixer\FixerInterface $fixer, SplFileInfo $file): ?PhpCsFixer\Fixer\FixerInterface {
+            // disable the 'strict_param' rule for all files in the "tests" directory
+            if ('strict_param' === $fixer->getName() && strpos($file->getPathname(), '/tests/') !== false) {
+                return null;
+            }
+            // reconfigure the 'array_syntax' rule to use long syntax for all files in the "bin" directory
+            if ('array_syntax' === $fixer->getName() && strpos($file->getPathname(), '/bin/') !== false) {
+                $fixer = clone $fixer;
+                $fixer->configure(['syntax' => 'long']);
+            }
```

Comment:
I'm not convinced to reconfigure fixers on the fly (also easy to forget to clone the fixer). Filter should filter-out.

I like the idea to unify codebase, it's OKish to have exception and not apply given rule, but applying different configuration based on path is a different beast

Reply:
> I'm not convinced to reconfigure fixers on the fly (also easy to forget to clone the fixer)

What about adding a note about that in the docs? See 4962436143fd49619d69f8bf45f925a22fe6864d